apiserver/facades/agent/uniter/uniter.go

```diff
@@ -89,6 +92,10 @@ type UniterAPI struct {
 	logger corelogger.Logger
 }
 
+type UniterAPIv20 struct {
+	*UniterAPI
+}
```

Comment:
issue: The UniterAPI is always the most current version of the facade. UniterAPIv19 will be the new struct which is the receiver for the Watch() method we're trying to deprecate. WatchApplication would have UniterAPI as its receiver, with the special case for UniterAPIv19 to indicate it's not available on that facade.

Checkout how it was done in the [ModelConfig facade](https://github.com/juju/juju/blob/79513a67e7e0f972b59613e35b1eba940c7f7486/apiserver/facades/client/modelconfig/register.go#L16-L32)

Reply:
Makes a lot of sense to do it that way, have changed it.